### ISSUE TITLE
onMouseLeave throws uncaught canceled

### ISSUE THREAD

**Issue by kellengreen** [microsoft/monaco-editor#2382]
I'm seeing uncaught errors when moving the cursor between the two halves of `createDiffEditor`.

**Steps to Reproduce:**

1.  Open up this HTML snippet, with dev tools running.
```html
<!DOCTYPE html>
<html>
  <body>
    <div style="height: 50vh"></div>
    <script>
      window.require = { paths: { vs: '/monaco/min/vs' } }
    </script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/monaco-editor/0.23.0/min/vs/loader.min.js"></script>
    <script>
      require(['vs/editor/editor.main'], () => {
        const elem = document.querySelector('div')
        const editor = monaco.editor.createDiffEditor(elem)
        const original = monaco.editor.createModel('foo')
        const modified = monaco.editor.createModel('bar')
        editor.setModel({ original, modified })
      })
    </script>
  </body>
</html>
```
2. Click anywhere on Monaco editor.
3. Move the mouse between the two editor halves.
4. You should see uncaught errors in the console log.  

**Error Output:**
```console
Uncaught (in promise) Canceled: Canceled
    at Object.d [as canceled] (/monaco/min/vs/editor/editor.main.js:8:711)
    at g.cancel (/monaco/min/vs/editor/editor.main.js:51:15831)
    at Function.onMouseLeave (/monaco/min/vs/editor/editor.main.js:103:10222)
    at /monaco/min/vs/editor/editor.main.js:103:5373
    at p.fire (/monaco/min/vs/editor/editor.main.js:51:1836)
    at HTMLDivElement.g (/monaco/min/vs/editor/editor.main.js:51:6378)
```

**Notes** 
* Reproduced in `0.22.x` and `0.23.0`.

**Comment by alexdima** [comment]
The error is harmless, the problem is that we have a rejected promise that is not explicitly handled and this makes it by default to the browser's console.

```
Uncaught (in promise) Canceled: Canceled
    at canceled (http://127.0.0.1:8080/vscode/out/vs/base/common/errors.js:109:23)
    at Delayer.cancel (http://127.0.0.1:8080/vscode/out/vs/base/common/async.js:244:57)
    at Function.onMouseLeave (http://127.0.0.1:8080/vscode/out/vs/base/browser/ui/sash/sash.js:291:31)
    at http://127.0.0.1:8080/vscode/out/vs/base/browser/ui/sash/sash.js:66:84
    at Emitter.fire (http://127.0.0.1:8080/vscode/out/vs/base/common/event.js:479:38)
    at HTMLDivElement.fn (http://127.0.0.1:8080/vscode/out/vs/base/browser/event.js:10:35)
```

**Comment by kellengreen** [comment]
Harmless for end users perhaps. 

However when running a react app in development the whole site will halt to capture this error.

Can we capture the rejection to prevent this?

![image](https://user-images.githubusercontent.com/1750507/110790986-97822d80-8269-11eb-84e6-70ec5dad851c.png)

**Comment by alexdima** [comment]
Yes, I think browsers have a global unhandled promise error handler, which we make use of in vscode, but the monaco editor cannot do this by its own, since that would be stepping on the embedder's toes.

The fix I pushed is to handle the error by adding a no-op error handler to the promise.

I can try looking up what the global unhandled promise error handler is called.

**Comment by alexdima** [comment]
Found it. You could do the following in your app:

```js
window.addEventListener('unhandledrejection', function(event) {
	if (event.reason && event.reason.name === 'Canceled') {
		// monaco editor promise cancelation
		event.preventDefault();
	}
});
```

**Comment by joaomoreno** [comment]
Yeah too bad we can't tell the browser which errors are fine to be thrown by unhandled promise rejections, ie cancellations.

**Comment by kellengreen** [comment]
Shame this can't be addressed.

For anyone coming across this for a solution. The above `unhandledrejection` workaround will require `crossorigin="anonymous"` on your `<script>` if the Monaco files are located on another origin.

**Comment by everhardt** [comment]
Does it make sense that I file this bug in the vscode repo? Or is it required that this error ends up in the global unhandled promise error handler in VS Code? 

The reason I'm asking: I already have two such error suppressors in my code (for #1937 and #1842, which might not be needed anymore as I see they're closed, but still) and am now adding a third. It's not really nice :)

**Comment by kellengreen** [comment]
Perhaps? It seems reasonable to not have a library throw errors into global space.

**Comment by alexdima** [comment]
> Shame this can't be addressed.

🤷‍♂️ . This was fixed in https://github.com/microsoft/vscode/commit/49cad9a1c0d9ef01d66eef60b261c7ebcffcef23 , so it was addressed.

**Comment by kellengreen** [comment]
Oh, I didn't catch that, never mind then.

**Comment by everhardt** [comment]
I missed that as well. Thanks a lot!

**Comment by jquintozamora** [comment]
should this issue be fixed in monaco-editor 0.24 ?

**Comment by kellengreen** [comment]
I'm not able to reproduce in `0.24.0` 👍
